pipeline/utils/downloads.py

```diff
@@ -0,0 +1,92 @@
+import io
+import json
+import os
+import shutil
+
+from google.api_core.exceptions import RequestRangeNotSatisfiable
+from google.cloud import storage as RealStorage
+
+from pipeline.utils.logging import get_logger
+
+google_cloud_storage: RealStorage
+
+
+# Use mocked downloads if they exist.
+if os.environ.get("MOCKED_DOWNLOADS"):
+    logger = get_logger("google_cloud(mocked)")
+
+    class MockedBlob:
```

Comment:
The best practice in python is to implement all the mocking using various libraries. For example: https://docs.python.org/3/library/unittest.mock.html
https://pypi.org/project/pytest-mock/

Python is really easy to monkey patch, so usually you do that and use those libs right in the tests without adding complexity to the functional code. I suggest migrating all the new mocks to this approach.

Then you wouldn't need this file at all and the unnecessary dependencies to `utils` would go away.

Reply:
So using `run_task` as an abstraction means that we can't use monkey patching like you are suggesting, as these are invoked via bash. I'm keeping some `os.environ` code for looking into mocked downloads, but I'm keeping that logic isolated into a module. It feels a bit simpler without the google GCS module.